lib/backend/firestore/firestorebk.go

```diff
@@ -709,19 +709,38 @@ func (b *Backend) purgeExpiredDocuments() error {
 // deleteDocuments removes documents from firestore in batches to stay within the
 // firestore write limits
 func (b *Backend) deleteDocuments(docs []*firestore.DocumentSnapshot) error {
-	for i := 0; i < len(docs); i += commitLimit {
-		batch := b.svc.Batch()
+	mDocs := make(map[string]*firestore.DocumentSnapshot)
+	for _, doc := range docs {
+		mDocs[doc.Ref.Path] = doc
+	}
 
-		for j := 0; j < commitLimit && i+j < len(docs); j++ {
-			batch.Delete(docs[i+j].Ref)
-		}
+	batch := b.svc.BulkWriter(b.clientContext)
+	jobs := make([]*firestore.BulkWriterJob, 0, len(mDocs))
 
-		if _, err := batch.Commit(b.clientContext); err != nil {
+	i := 0
+	for _, doc := range docs {
+		job, err := batch.Delete(doc.Ref)
+		if err != nil {
 			return ConvertGRPCError(err)
```

Comment:
If you return early you probably need to clear the batch otherwise it will stay open forever?
I don't know the specific behavior of firestore on these cases

Reply:
From what I understand, this is not the case since we don't use the transaction API which had similar semantics to the old batch API.

Reply:
got it